ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -53,22 +54,39 @@ class ComponentTransportLocalAssemblerInterface
       public NumLib::ExtrapolatableElement
 {
 public:
+    ComponentTransportLocalAssemblerInterface() : _coupled_solutions(nullptr) {}
+
+    void setStaggeredCoupledSolutions(
+        std::size_t const /*mesh_item_id*/,
+        CoupledSolutionsForStaggeredScheme* const coupling_term)
+    {
+        _coupled_solutions = coupling_term;
+    }
+
     virtual std::vector<double> const& getIntPtDarcyVelocity(
         const double t,
         GlobalVector const& current_solution,
         NumLib::LocalToGlobalIndexMap const& dof_table,
         std::vector<double>& cache) const = 0;
+
+protected:
+    CoupledSolutionsForStaggeredScheme* _coupled_solutions;
 };
 
 template <typename ShapeFunction, typename IntegrationMethod,
           unsigned GlobalDim>
 class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
 {
-    // Pressure always first
-    static const int pressure_index = 0;
+    // Pressure
+    static const int hydraulic_process_id = 0;  // For staggered scheme
+    static const int pressure_index = 0;        // For monolithic scheme
     static const int pressure_size = ShapeFunction::NPOINTS;
-    // Size of concentration to each component
-    static const int concentration_size = ShapeFunction::NPOINTS;
+    // Concentration
+    static const int first_transport_process_id = 1;  // For staggered scheme
```

Comment:
The name implies there is a second transport process, isn't it?

Reply:
I intend to make this variable different from another variable 'transport_process_id'. This variable is only used in calculation of fluid density which is herein assumed to be dependent upon the concentration of 1st component.